File: app/src/main/java/com/odnovolov/randomuser/screen/userprofile/UserProfileView.java
package com.odnovolov.randomuser.screen.userprofile;

import com.odnovolov.randomuser.model.User;
import moxy.MvpView;
import moxy.viewstate.strategy.AddToEndSingleTagStrategy;
import moxy.viewstate.strategy.StateStrategyType;

public interface UserProfileView extends MvpView {
    @StateStrategyType(value = AddToEndSingleTagStrategy.class, tag = "status")
    void showUserProfile(User user);

    @StateStrategyType(value = AddToEndSingleTagStrategy.class, tag = "status")
    void showProgress();

    @StateStrategyType(value = AddToEndSingleTagStrategy.class, tag = "status")
    void showError();
}

File: app/src/main/java/com/odnovolov/randomuser/screen/userprofile/UserProfileView.kt
package com.odnovolov.randomuser.screen.userprofile

import com.odnovolov.randomuser.model.User
import moxy.MvpView
import moxy.viewstate.strategy.AddToEndSingleTagStrategy
import moxy.viewstate.strategy.StateStrategyType

interface UserProfileView : MvpView {
    @StateStrategyType(value = AddToEndSingleTagStrategy::class, tag = "status")
    fun showUserProfile(user: User)

    @StateStrategyType(value = AddToEndSingleTagStrategy::class, tag = "status")
    fun showProgress()

    @StateStrategyType(value = AddToEndSingleTagStrategy::class, tag = "status")
    fun showError()
}